mithril-common/src/cardano_transaction_parser.rs

```diff
@@ -115,60 +116,61 @@ impl CardanoTransactionParser {
 
     /// Read blocks from immutable file
     fn read_blocks_from_immutable_file(immutable_file: &ImmutableFile) -> StdResult<Vec<Block>> {
-        let hardano_blocks = hardano_blocks(immutable_file)?;
-
-        let mut blocks = Vec::new();
-        for block in hardano_blocks {
-            let block = block.map_err(|e| {
-                anyhow!(e).context(format!(
-                    "Error while reading block in immutable file: '{:?}'",
-                    immutable_file.path
-                ))
+        let cardano_blocks_reader =
+            CardanoTransactionParser::cardano_blocks_reader(immutable_file)?;
+
+        cardano_blocks_reader
+            .into_iter()
+            .map(|b| CardanoTransactionParser::convert_to_block(b, immutable_file))
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| anyhow!(e))
```

Comment:
This looks a bit unconventional as the collected type should be `Vec<StdResult<_>>` instead of `StdResult<Vec<_>>` :thinking:  Maybe a for loop is better here?